From 0994eb5433e26b3852a57c3eacdad954e1b9fbdb Mon Sep 17 00:00:00 2001
From: Murala S Swamy <ssmurala@habana.ai>
Date: Tue, 10 Jun 2025 11:00:32 +0300
Subject: [PATCH] [JSW-670] jgs: enable palad boot loader

Palad boot loader requires misc arc iccm and
dccm details, those are added as part of this
patch.

Change-Id: Ia0acbb0c6dec9c589b5f9d2d4f29c733b2281c15
---

diff --git a/fs/jgs/address_space.h b/fs/jgs/address_space.h
index d62df51..9f6d866 100644
--- a/fs/jgs/address_space.h
+++ b/fs/jgs/address_space.h
@@ -113,6 +113,7 @@
 #define IO_0_MNG_ARC_ICCM_BASE		(mmIO_0_CFG_FW_MEM_MNG_ARC_ICCM_BASE)
 #define IO_0_PID_ARC_ICCM_BASE		(mmIO_0_CFG_FW_MEM_PID_ARC_ICCM_BASE)
 #define IO_0_MISC_ARC_ICCM_BASE		(mmIO_0_CFG_FW_MEM_MISC_ARC_ICCM_BASE)
+#define IO_0_MISC_ARC_DCCM_BASE		(mmIO_0_CFG_FW_MEM_MISC_ARC_DCCM_BASE)
 
 #define IO_0_SP_SRAM_BASE_FULL		(GET_FULL_ADDR(mmIO_0_CFG_SP_RAM_BASE)) /* TODO: SOL Update */
 #define IO_0_FW_MEM_BASE_FULL		(GET_FULL_ADDR(mmIO_0_CFG_FW_MEM_BASE))
@@ -122,7 +123,7 @@
 #define IO_0_MNG_ARC_ICCM_BASE_FULL	(GET_FULL_ADDR(mmIO_0_CFG_FW_MEM_MNG_ARC_ICCM_BASE))
 #define IO_0_PID_ARC_ICCM_BASE_FULL	(GET_FULL_ADDR(mmIO_0_CFG_FW_MEM_PID_ARC_ICCM_BASE))
 #define IO_0_MISC_ARC_ICCM_BASE_FULL	(GET_FULL_ADDR(mmIO_0_CFG_FW_MEM_MISC_ARC_ICCM_BASE))
-
+#define IO_0_MISC_ARC_DCCM_BASE_FULL	(GET_FULL_ADDR(mmIO_0_CFG_FW_MEM_MISC_ARC_DCCM_BASE))
 
 /* TODO: why this is not working for JGS? */
 #define CONV_64B_TO_ARC_FW_MEM(addr)	((addr & 0x3FFFFF) + mmIO_0_CFG_FW_MEM_BASE)
